### PR TITLE
asadiqbal08/Migrating from ElasticSearch to OpenSearch

### PR DESCRIPTION
**Pull request by asadiqbal08**
#### What are the relevant tickets?
fixes: https://github.com/mitodl/micromasters/issues/5155

#### What's this PR do?
We migrated from Elasticsearch to OpenSearch

#### How to test ?
- Make sure the env for `OPENSEARCH_INDEX` is set e.g. `micromaster`, also the `OPENSEARCH_DEFAULT_PAGE_SIZE=50`
- Build the container with OpenSearch image
- Execute the `recreate_index` command to check the indexing pattern.
- Other, test the searching via providing some queryparams to OpenSearch searching endpoint `/api/v0/search/`

### CODE REVIEW

**Comment by pdpinch** [comment]
It's going to be at least a few days before @mitodl/devops will be able to set up a opensearch cluster for you to test with. In the meantime can you help them with some more information:

- What API endpoints does this app use?
- What are the names of the indexes that we will need?

This will help them with setting up permissions and access controls.

**Comment by asadiqbal08** [comment]
> What API endpoints does this app use?

@pdpinch If I looked around the codebase then these [URL](https://github.com/mitodl/micromasters/blob/master/search/urls.py#L8) is used along with ElasticSearch query data to get results. As an example:

screencapture-mm-odl-local-8079-api-v0-search-2022-02-09-20_40_48

Screenshot 2022-02-09 at 9 10 45 PM


> What are the names of the indexes that we will need?

I used this setting locally for `OPENSEARCH_INDEX='micromaster'` and [this](https://github.com/mitodl/micromasters/blob/master/search/connection.py#L97-L112) is basically making alias names but I believe DevOps have to look around these settings while configuring over Heruko.

```
HEROKU_PARENT_APP_NAME
HEROKU_APP_NAME
```

**Comment by pdpinch** [comment]
I think @blarghmatey wanted to know the elasticsearch endpoints that the MIcroMasters app hits. One of his bigger tasks will be updating the access controls around those endpoints, because that's different in the AWS service than in our own cluster.

**Comment by Ardiea** [comment]
> I think @blarghmatey wanted to know the elasticsearch endpoints that the MIcroMasters app hits. One of his bigger tasks will be updating the access controls around those endpoints, because that's different in the AWS service than in our own cluster.

This would still be very good information to have in the near-ish future. I'm starting to look at spinning up opensearch clusters for you this week/next week, I believe. I opened https://github.com/mitodl/ol-infrastructure/issues/639 to track the work on the devops side.

**Comment by Ardiea** [comment]
@asadiqbal08 

https://github.com/mitodl/ol-infrastructure/pull/647

I have created these OpenSearch clusters in CI, QA, and Production. I have written the code to hook the Read-Only IAM policy up to MIT-Open but I have not run it and I have not written anything else (I think only micromasters remains?). If you need the details on the policy ARNs I can get hose for you. LMK how I can help further!

**Comment by asadiqbal08** [comment]
@Ardiea can we deploy this PR changes to CI environment to test the OpenSearch functional behavior and its searching endpoints ?

**Comment by pdpinch** [comment]
@asadiqbal08 can you rebase this?

I tried clicking on "Create Review app" in the Heroku dashboard for you, but I got the following errors:

```
"OPENSEARCH_HTTP_AUTH" is required, config var "OPENSEARCH_SHARD_COUNT" is required, config var "OPENSEARCH_INDEX" is required, config var "OPENSEARCH_URL" is required • View log 
```

**Comment by asadiqbal08** [comment]
Rebased. I guess you need to set these vars. Devops can help out about them. 

```
ELASTICSEARCH_INDEX=`micromaster`
OPENSEARCH_URL=https://(your_open_search_url ) for connecting to OpenSearch cluster.
OPENSEARCH_HTTP_AUTH=(your_cluster_name):(key) (Basic auth settings for connecting to Opensearch)
```

File: search/indexing_api_test.py [hunk 1]
```diff
@@ -11,7 +11,9 @@
 )
 from django.conf import settings
 from django.db.models.signals import post_save
-from elasticsearch.exceptions import NotFoundError
+from django.test import override_settings
+# from elasticsearch.exceptions import NotFoundError
```
Comment (annagav) [inline search/indexing_api_test.py:15]: could you remove this

**Review by annagav — CHANGES_REQUESTED** [review]
Code looks good, just a minor comment. Going to test functionality

**Comment by annagav** [comment]
Functionality looks good 👍

**Comment by asadiqbal08** [comment]
@annagav should I merge it ?

**Review by annagav — APPROVED** [review]
👍

**Pull request merged**